interfaces/builtin/ofono_test.go

```diff
@@ -0,0 +1,213 @@
+			Apps:      map[string]*snap.AppInfo{"app": app},
+		},
+	}
+	release.OnClassic = false
+	snippet, err := s.iface.ConnectedPlugSnippet(s.plug, slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(string(snippet), testutil.Contains, `peer=(label="snap.ofono.app"),`)
+}
+
+func (s *OfonoInterfaceSuite) TestConnectedPlugSnippetUsesUnconfinedLabelOnClassic(c *C) {
+	release.OnClassic = true
+	snippet, err := s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	// verify apparmor connected
+	c.Assert(string(snippet), testutil.Contains, "#include <abstractions/dbus-strict>")
+	// verify classic connected
+	c.Assert(string(snippet), testutil.Contains, "peer=(label=unconfined),")
+}
+
+func (s *OfonoInterfaceSuite) TestConnectedPlugSnippetAppArmor(c *C) {
+	release.OnClassic = false
+	snippet, err := s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	// verify apparmor connected
+	c.Assert(string(snippet), testutil.Contains, "#include <abstractions/dbus-strict>")
+	// verify classic didn't connect
+	c.Assert(string(snippet), Not(testutil.Contains), "peer=(label=unconfined),")
+}
+
+func (s *OfonoInterfaceSuite) TestConnectedPlugSnippetSecComp(c *C) {
+	snippet, err := s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecuritySecComp)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+
+	c.Check(string(snippet), testutil.Contains, "getsockname\n")
+}
+
+func (s *OfonoInterfaceSuite) TestConnectedSlotSnippetAppArmor(c *C) {
+	snippet, err := s.iface.ConnectedSlotSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+
+	c.Check(string(snippet), testutil.Contains, "interface=org.ofono.*\n")
+}
+
+func (s *OfonoInterfaceSuite) TestConnectedSlotSnippetSecComp(c *C) {
+	snippet, err := s.iface.ConnectedSlotSnippet(s.plug, s.slot, interfaces.SecuritySecComp)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, IsNil)
+}
+
+func (s *OfonoInterfaceSuite) TestPermanentPlugSnippetAppArmor(c *C) {
+	snippet, err := s.iface.PermanentPlugSnippet(s.plug, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, IsNil)
+}
+
+func (s *OfonoInterfaceSuite) TestPermanentPlugSnippetSecComp(c *C) {
+	snippet, err := s.iface.PermanentPlugSnippet(s.plug, interfaces.SecuritySecComp)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, IsNil)
+}
+
+func (s *OfonoInterfaceSuite) TestPermanentSlotSnippetAppArmor(c *C) {
+	snippet, err := s.iface.PermanentSlotSnippet(s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+
+	c.Check(string(snippet), testutil.Contains, "/dev/net/tun rw,")
+}
+
+func (s *OfonoInterfaceSuite) TestPermanentSlotSnippetSecComp(c *C) {
+	snippet, err := s.iface.PermanentSlotSnippet(s.slot, interfaces.SecuritySecComp)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+
+	c.Check(string(snippet), testutil.Contains, "setsockopt\n")
```

Comment:
Please add: `c.Check(string(snippet), testutil.Contains, "listen\n")` (`setsockopt` is in default policy.